standard_rules/unusual_login.py

```diff
@@ -0,0 +1,73 @@
+import json
+import requests
+import panther_event_type_helpers as event_type
+from panther_oss_helpers import get_string_set, put_string_set
+FINGERPRINT_THRESHOLD = 3
+
+
+def rule(event):
+    # Pre-filter to save compute time where possible. event_type_id = 5 is login events.
+    if not event.udm('event_type') or event.udm(
+            'event_type') != event_type.SUCCESSFUL_LOGIN:
+        return False
+    # we use udm 'actor_user' field as a key
+    if not event.udm('actor_user'):
+        return False
+
+    # Lookup geo-ip data via API call
+    url = 'https://ipinfo.io/' + event.udm('source_ip') + '/geo'
+
+    # Skip API call if this is a unit test
+    if __name__ == 'PolicyApiTestingPolicy':
+        resp = lambda: None
+        setattr(resp, 'status_code', 200)
+        setattr(resp, 'text', event['api_data'])
+    else:
+        # This response looks like the following:
+        # {‘ip': '8.8.8.8', 'city': 'Mountain View', 'region': 'California', 'country': 'US',
+        # 'loc': '37.4056,-122.0775', 'postal': '94043', 'timezone': 'America/Los_Angeles'}
+        resp = requests.get(url)
+
+    if resp.status_code != 200:
+        # Could raise an exception here for ops team to look into
+        return False
```

Comment:
Now that rule erroring is forwarded to users, lets make this do something like `raise Exception("API call failed: GET {} returned {}".format(url, resp.status_code))`